### PR TITLE
Role change fix

### PR DESCRIPTION
**Pull request by Undid-Iridium**
Role change fix

### CODE REVIEW

**Review by Thundermaker300 — APPROVED** [review]

File: Exiled.Events/EventArgs/Player/ChangingRoleEventArgs.cs [hunk 1]
```diff
@@ -104,6 +104,7 @@ public bool ShouldPreserveInventory
         /// </summary>
         public SpawnReason Reason { get; set; }
 
+
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Player/ChangingRoleEventArgs.cs:107]: Not needed

**Comment by Thundermaker300** [comment]
Fix build error :)

File: Exiled.Events/Patches/Events/Player/ChangingRole.cs [hunk 2]
```diff
@@ -193,16 +178,20 @@ private static void UpdatePlayerRole(RoleTypeId newRole, API.Features.Player pla
...
         }
 
-        private static void ChangeInventory(API.Features.Player player, List<ItemType> items, Dictionary<ItemType, ushort> ammo, RoleTypeId prevRole, RoleTypeId newRole, RoleChangeReason reason)
+        private static void ChangeInventory(ChangingRoleEventArgs ev, RoleTypeId prevRole)
         {
+            API.Features.Player player = ev.Player;
+            List<ItemType> items = ev.Items;
+            Dictionary<ItemType, ushort> ammo = ev.Ammo;
+            RoleChangeReason reason = (RoleChangeReason)ev.Reason;
+            RoleTypeId newRole = ev.NewRole;
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/ChangingRole.cs:187]: You should just use `ev.X` below instead of making new vars.

**Review by iopietro — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Events/Player/ChangingRole.cs (hunk 2 above)
Comment (Undid-Iridium) [reply Exiled.Events/Patches/Events/Player/ChangingRole.cs:187]: I was trying to be quick but sure

**Review by iopietro — APPROVED** [review]

**Review by Thundermaker300 — APPROVED** [review]

**Review by iopietro — APPROVED** [review]

**Pull request merged**